README.md

```diff
@@ -26,30 +23,130 @@ dune-release help release # for help about releasing your package
 dune-release help         # for more help
 ```
 
-[doc]: https://samoht.github.com/dune-release/
+Calling `dune-release` without any argument will start the automated release process, composed of the following steps:
+- create the distribution archive;
+- publish it online with its documentation;
+- create an opam package;
+- submit it to OCaml's opam repository.
+
+The most basic workflow is:
+
+```
+dune-release tag
+dune-release
+```
+
+Each step is refined and explained with more details below.
+If this is your first time using `dune-release` you might choose to run the `dune-release` commands with the `--dry-run` argument, so that no action is actually performed but it allows you to check what would be done. And then run the `dune-release` commands without `--dry-run` when you are sure of what you are doing.
+
+
+### Tag the distribution with a version
+
+This step should be executed prior to running the `dune-release` automated process.
+
+The tagging command of `dune-release` will extract the latest version tag from the package's change log and tag the VCS HEAD commit with it if it is invoked without argument:
+
+```
+dune-release tag
+```
+
+This will only work if the change log follows a certain format. If you do not want to rely on this extraction you can specify it on the command line:
+
+```
+dune-release tag v1.0.1
+```
+
+You can also directly use your VCS instead if the `dune-release tag` command does not fit your needs.
+
+If you need to delete a tag created by `dune-release`, use the following command:
+
+```
+dune-release tag -d v1.0.1
+```
+
+The full documentation of this command is available with
+```
+dune-release help tag
+```
+
+
+### Create the distribution archive
+
+Now that the release is tagged in your VCS, generate a distribution archive for it in the build directory with:
+
+```
+dune-release distrib
+```
+
+This uses the source tree of the HEAD commit for creating a distribution in the build directory. The distribution version string is the VCS tag description (e.g. `git-describe`) of the HEAD commit. Alternatively it can be specified on the command line.
+
+Basic checks are performed on the distribution archive when it is created, but save time by catching errors early. Hence test that your source repository lints and that it builds in the current build environment and that the package tests pass.
+
+```
+dune-release lint
+dune build # Check out the generated opam install file too
+dune runtest
+```
+
+The full documentation of this command is available with
+```
+dune-release help distrib
+```
+
 
-For instance:
+### Publish the distribution and documentation online
 
+Once the distribution archive is created you can now publish it and its documentation online.
```

Comment:
Probably worth explaining what publishing means here, ie:
- for the distrib it means: creating a Github release associated with the tag and upload the distrib tarball as a release artifact
- for the doc, publishing the dune generated documentation to gh-pages to be served as a static website on github.io

Reply:
done